docs: Service Registry and Service Binding docs (#299)
redhat-developer/app-services-operator: docs/service_binding.adoc

== Service Binding 

> When you are writing an application that requires multiple services (such as Kafka and a Service Registry), you normally have to configure connections to these services by hand in a configuration file. These connections often include public data such as well known urls as well as private data such as passwords and account IDs. Custom resources managed by the RHOAS operator provide service binding annotations which allows the link:{https://redhat-developer.github.io/service-binding-operator/userguide/intro.html}[Service Binding Operator (SBO)] to expose this connection information to your application. Because the SBO manages this information it can stay securely inside of your Open Shift cluster instead of being shared around in configuration files in source control.

Reviewer: ```suggestion
When you are writing an application that requires multiple services (such as Kafka and a Service Registry), you normally have to configure connections to these services by hand in a configuration file. These connections often include public data such as well known urls as well as private data such as passwords and account IDs. Custom resources managed by the RHOAS operator provide service binding annotations which allows the link:{https://redhat-developer.github.io/service-binding-operator/userguide/intro.html}[Service Binding Operator (SBO)] to expose this connection information to your application. Because the SBO manages this information it can stay securely inside of your Open Shift cluster instead of being shared across different configuration files in source control.
```

    ## SASL PLAIN credentials
    service.binding/user: >-
      path={.status.serviceAccountSecretName},objectType=Secret,sourceKey=client-id
    service.binding/password: >-
      path={.status.serviceAccountSecretName},objectType=Secret,sourceKey=client-secret
    ## SASL OAUTH BEARER credentials
    service.binding/clientId: >-
           path={.status.serviceAccountSecretName},objectType=Secret,sourceKey=client-id
    service.binding/clientSecret: >-
      path={.status.serviceAccountSecretName},objectType=Secret,sourceKey=client-secret
    service.binding/saslMechanism: 'path={.status.metadata.saslMechanism}'
    service.binding/securityProtocol: 'path={.status.metadata.securityProtocol}'
    ## Additional binding metadata required for Quarkus
    service.binding/type: 'path={.status.metadata.type}'
    service.binding/provider: 'path={.status.metadata.provider}'
----

Similarly to the `ServiceRegistryConnection` annotations, these allow the SBO to create the bindings `bootstrapServers`, `user`, `password`, `clientId`, `clientSecret`, `saslMechanism`, `securityProtocol`, `type`, and `provider` from values provided on KafkaConnection instances.

=== Binding Example

> If you have a deployment named `kafka-avro-schema-quickstart-app` and a ServiceRegistryConnection `quickstart-service-registry` the following ServiceBinding resource can be used to bind the Service Registry connection information into the pods of the deployment. The application in the deployment may look these properties up as values on the filesystem, or use a quarkus extension to automatically consume them in their application.

Reviewer: ```suggestion
For a deployment named `kafka-avro-schema-quickstart-app` and a ServiceRegistryConnection `quickstart-service-registry` the following ServiceBinding resource can be used to bind the Service Registry connection information into the pods of the deployment. The application in the deployment may look these properties up as values on the filesystem, or use a quarkus extension to automatically consume them in their application.
```
